### PR TITLE
Fix partition root's reloptions missing when created in GPDB syntax

### PR DESCRIPTION
**Pull request by huansong**
Starting from 0ae9dce7b54f313b88c8306d0640cb45fbebb34c the partition root can carry reloptions as well. However, we missed removing a prior logic where we used to avoid setting the reloptions field for partition root when the partition is created in GPDB-style syntax. E.g. when we do:
```
CREATE TABLE part(a int) PARTITION BY RANGE(a) (PARTITION part1 START (1) END (100)) WITH (fillfactor=70);
```
The partition root will not have the reloptions but the child will.

Fixing that by removing that logic, and fix/add a few test cases:
  * AOCO_Compression: new added partition should use the given
    reloptions, and shouldn't inherit parent's reloptions if
    parent has different AM.
  * partition: partition root should carry reloptions correctly.
  * partition_storage: newly added partition should inherit parent's reloptions.

In addition, a newly created child partition shouldn't inherit its parent's
reloptions if they have different access methods.

P.S. similarly, we supported setting access method for partition root in afde39b3f973. There was a prior logic removing the AM field of the partition root, and we got rid of that logic in the aforementioned commit.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/commands/tablecmds.c [hunk 1]
```diff
@@ -790,7 +790,10 @@ DefineRelation(CreateStmt *stmt, char relkind, Oid ownerId,
...
 		accessMethodId = get_table_am_oid(default_table_access_method, false);
 
-	/* GPDB: for partitioned tables, inherit reloptions from the parent. */
+	/* 
+	 * GPDB: for partitioned tables, inherit reloptions from the parent. 
+	 * Note this is applicable only if the parent has the same AM as the child.
+	 */
 	if (stmt->partbound && (relkind == RELKIND_RELATION || relkind == RELKIND_PARTITIONED_TABLE))
 	{
 		Oid			relid;
```
Comment (soumyadeep2007) [inline src/backend/commands/tablecmds.c:799]: Reading this makes me feel that we should rename `rel, relid` as `parentrel, parentrelid`?

**Review by soumyadeep2007 — APPROVED** [review]
LGTM!

**Comment by huansong** [comment]
There was a CI failure in pg_restore test that it couldn't create a partitioned table with the reloption `checksum`. It complained that `checksum` isn't compatible with heap table. This is because pg_dump didn't set proper access method (which is AO in this case) for that table. Fixing that now. That was broken since afde39b3f973c429aa4175e5d70575566bb905bd where we supported AM for partition root, but only exposed now because the reloption change in this PR.

Thanks a lot to @bmdoil 's help with pg_dump/restore.

**Review by divyeshddv — APPROVED** [review]
LGTM!

**Pull request merged**